### PR TITLE
Cleanup dependencies and fix cron job

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind documentation
/kind testing

#### What this PR does / why we need it:
Cleanup dependencies and fix cron job

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by MartinForReal** [comment]
<html>
<body>


/test pull-cloud-provider-azure-e2e-capz
--


<br class="Apple-interchange-newline">
</body>
</html>

File: .github/workflows/hugo.yml [hunk 1]
```diff
@@ -56,8 +56,8 @@ jobs:
         run: |
           cd site
           hugo \
-            --minify \
-            --baseURL "${{ steps.pages.outputs.base_url }}/"
```
Comment (feiskyer) [inline .github/workflows/hugo.yml:60]: baseURL is required

Comment (MartinForReal) [reply .github/workflows/hugo.yml:60]: yes. I've added two environment config for baseurl.

Comment (MartinForReal) [reply .github/workflows/hugo.yml:60]: hugo will load config under githubpage when --environment githubpage is specified. 
and config.toml under githubpage will set baseurl to  "https://kubernetes-sigs.github.io/cloud-provider-azure/"

**Pull request merged**